syntax = "proto3";

package task;

import "google/protobuf/timestamp.proto";
import "google/protobuf/empty.proto";

service TaskService {
  // 태스크 생성
  rpc CreateTask (CreateTaskRequest) returns (TaskResponse);
  
  // 태스크 목록 조회
  rpc ListTasks (ListTasksRequest) returns (ListTasksResponse);
  
  // 특정 태스크 조회
  rpc GetTask (GetTaskRequest) returns (TaskResponse);
  
  // 태스크 업데이트
  rpc UpdateTask (UpdateTaskRequest) returns (TaskResponse);
  
  // 태스크 삭제
  rpc DeleteTask (DeleteTaskRequest) returns (DeleteTaskResponse);
}

message CreateTaskRequest {
  string title = 1;
  optional string description = 2;
}

message ListTasksRequest {
  int32 skip = 1;
  int32 limit = 2;
}

message ListTasksResponse {
  repeated TaskResponse tasks = 1;
}

message GetTaskRequest {
  int32 task_id = 1;
}

message UpdateTaskRequest {
  int32 task_id = 1;
  optional string title = 2;
  optional string description = 3;
  optional bool completed = 4;
}

message DeleteTaskRequest {
  int32 task_id = 1;
}

message DeleteTaskResponse {
  bool success = 1;
}

message TaskResponse {
  int32 id = 1;
  string title = 2;
  optional string description = 3;
  bool completed = 4;
  google.protobuf.Timestamp created_at = 5;
  optional google.protobuf.Timestamp updated_at = 6;
} 